kie-server-parent/kie-server-services/kie-server-services-task-assigning/kie-server-services-task-assigning-core/src/test/java/org/kie/server/services/taskassigning/core/model/solver/realtime/AbstractProblemFactChangeTest.java

```diff
@@ -35,14 +35,22 @@
 public abstract class AbstractProblemFactChangeTest extends AbstractTaskAssigningCoreTest {
 
     final Random random = new Random();
+    private static int changeIds = 1;
```

Comment:
Suggestion: instead of synchronizing access to this variable, can we use the `AtomicInteger`?